libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -758,6 +767,25 @@ void AP_GPS::update_instance(uint8_t instance)
     }
 #endif
 
+    if (data_should_be_logged) {
+        // keep count of delayed frames and average frame delay for health reporting
+        const uint16_t gps_max_delta_ms = 245; // 200 ms (5Hz) + 45 ms buffer
+        GPS_timing &t = timing[instance];
+
+        if (t.delta_time_ms > gps_max_delta_ms) {
+            t.delayed_count++;
+        } else {
+            t.delayed_count = 0;
+        }
+        if (t.delta_time_ms < 2000) {
+            if (t.average_delta_ms <= 0) {
+                t.average_delta_ms = t.delta_time_ms;
+            } else {
+                t.average_delta_ms = 0.98 * t.average_delta_ms + 0.02 * t.delta_time_ms;
```

Comment:
Probably worth adding the `f` specifier as the GPS compile units actually do use double math, just don't want to promote here by accident.

Reply:
done, although it doesn't matter as only SBF and GSOF backends use double